waiter/src/waiter/scheduler.clj

```diff
@@ -730,13 +735,25 @@
 
 (defn add-instance-to-buffered-collection!
   "Helper function to add/remove entries into the transient store"
-  [transient-store max-instances-to-keep service-id instance-entry initial-value-fn remove-fn]
```

Comment:
Why did we reorder the argument? It has triggered a bunch of changes in the callers.

Reply:
I reordered the argument because I wanted to use & args in the `add-instance-to-transient-store!` without argument reordering it would be less idiomatic:

```
(defn add-instance-to-transient-store!
  [transient-store service-id instance event-type & args]
  (log-service-instance instance event-type)
  (apply add-instance-to-buffered-collection! transient-store service-id instance args))
```

Reply:
If we don't have the override we don't have a need to use `args`.

Reply:
Please do not change the signature of this function, `max-instances-to-keep` should be passed by the caller.

Reply:
Resolved

Reply:
I still see a diff. There is no need for the extra set of prenthesis since we have only one signature for the function now.